### PR TITLE
Roomba roulette

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Clicking on roomba with harm intent and fun config allowed will have a 50% chance of deleting it.
And a 50% change of deleting you.

## Why It's Good For The Game

This thing is stealing my stuff, we need a way to kill it

## Changelog
:cl:
add: Roomba roulette. Just click on the roomba!
/:cl:

### CODE REVIEW

**Comment by Insane-Jimmy** [comment]
this is the best thing to do while im bored out of my mind in medbay while on crack

**Comment by Derrick2Electricboogaloo** [comment]
Oh god. FEAR THE ROOMBA

**Review by TiviPlus — APPROVED** [review]
lmao

**Comment by mathis650** [comment]
based and roombapilled

File: code/game/objects/machinery/roomba.dm [hunk 1]
```diff
@@ -49,6 +49,17 @@
 		return
 	step_to(src, get_step(src,newdir))
 
+/obj/machinery/roomba/attack_hand(mob/living/user)
+	if(!CONFIG_GET(flag/fun_allowed))
+		return
+	if(user.a_intent != INTENT_HARM)
+		return
+	tgui_alert(user, "Are you really sure to want to try your luck with the devilish roomba?", "The roomba rulette", list("Yes", "Yes!", "Yes?"))
```
Comment (BraveMole) [inline code/game/objects/machinery/roomba.dm:57]:
```suggestion
	tgui_alert(user, "Are you really sure to want to try your luck with the devilish roomba?", "The roomba roulette", list("Yes", "Yes!", "Yes?"))
```

**Comment by neohenry** [comment]
BUT DO WE RESPAWN?!

**Comment by lKiseki** [comment]
there is no mercy for those that would dare challenge the roomba

**Comment by token-map** [comment]
can't you just spam harm intent to get over the 50 percent chance

File: code/game/objects/machinery/roomba.dm [hunk 2]
```diff
@@ -49,6 +49,17 @@
...
+	if(!CONFIG_GET(flag/fun_allowed))
+		return
+	if(user.a_intent != INTENT_HARM)
+		return
+	tgui_alert(user, "Are you really sure to want to try your luck with the devilish roomba?", "The roomba roulette", list("Yes", "Yes!", "Yes?"))
+	if(prob(50))
+		explosion(user, 1, 0, 0, 0, 0, 4, "[user] lost at the roomba roulette")
+		return
+	explosion(src, 1, 0, 0, 0, 0, 4, "[user] won at the roomba roulette")
+	qdel(src)
```
Comment (SplinterGP) [inline code/game/objects/machinery/roomba.dm:62]: Imagine not turning the person into bread that the roomba will _**C O N S U M E**_

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/roomba.dm (hunk 2 above)
Comment (BraveMole) [reply code/game/objects/machinery/roomba.dm:62]: yeah ok

**Comment by AnCopper** [comment]
roomba_god

File: code/game/objects/machinery/roomba.dm (hunk 2 above)
Comment (BraveMole) [reply code/game/objects/machinery/roomba.dm:62]: Dis good enough for a shit post

**Pull request merged**